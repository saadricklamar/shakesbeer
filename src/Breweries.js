import React, { Component } from 'react';
import Beer from './Beer.js'
import './Breweries.scss';

class Breweries extends Component {
  constructor(props) {
    super(props);
    this.state = {
      beerList: []
    }

    this.toggleFavorite = this.toggleFavorite.bind(this);
  }

  componentWillMount() {
    this.setState({breweryName: this.props.name}, () => this.getBeers());
  }

  toggleBeerList = (e) => {
    const beerList = e.target.closest('.brewery-label').querySelector('.beer-dropdown');
    beerList.classList.toggle('hidden');
  }

  getBeers = () => {
    let match = this.props.dataset.find(brewery => {
      return this.props.id === brewery.FIELD1;
    });
    let beers = match && match.beers ? match.beers : null;
    this.setState({beerList: beers});
  }

  toggleFavorite = () => {
    !this.props.starredBreweries.includes(this.props.id) 
    ? this.props.updateStarredList(this.props.id, 'add')
    : this.props.updateStarredList(this.props.id)
  }
  
  render() {
    let favClass = this.props.starredBreweries.includes(this.props.id) 
    ? 'far fa-star fas' 
    : 'far fa-star';

    return (
      <div className='brewery-label'>
        <div className='brewery-header'>
          <i className={favClass} id='star-filler' onClick={this.toggleFavorite}></i>
          <h2 className='brewery-name' onClick={this.toggleBeerList}>{this.props.name}</h2>
        </div>
        <div className='beer-dropdown hidden'>
        {
        this.state.beerList ? (
          this.state.beerList.map(beer => {
            return <Beer beers={this.state.beerList}
                         beerName={beer.name}
                         key={beer.FIELD1}
                    />
          })
        ) : (null)
        }
        </div>
      </div>
    );
  }
}

export default Breweries;