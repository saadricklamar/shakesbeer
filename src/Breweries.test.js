import React from 'react';
import Breweries from './Breweries.js';
import { shallow } from 'enzyme';

const updateStarredListMock = jest.fn();
const mockData = [
    {
    "FIELD1": 90,
    "name": "The Lion Brewery",
    "city": "Wilkes-Barre",
    "state": "PA",
    "beers": [ {
        "FIELD1": 43,
        "abv": 0.045,
        "ibu": 18,
        "id": 2544,
        "name": "Send Help",
        "style": "American Blonde Ale",
        "brewery_id": 60,
        "ounces": 12
      },
      {
        "FIELD1": 44,
        "abv": 0.055,
        "ibu": null,
        "id": 2324,
        "name": "Cast Iron Oatmeal Brown",
        "style": "American Brown Ale",
        "brewery_id": 60,
        "ounces": 12
      },
      {
        "FIELD1": 45,
        "abv": 0.06,
        "ibu": null,
        "id": 2288,
        "name": "Reprise Centennial Red",
        "style": "American Amber / Red Ale",
        "brewery_id": 60,
        "ounces": 12
      },
      {
        "FIELD1": 46,
        "abv": 0.055,
        "ibu": null,
        "id": 2287,
        "name": "Alter Ego",
        "style": "American Black Ale",
        "brewery_id": 60,
        "ounces": 12
      }
    ]
  }
]

describe('Breweries', () => {
    let wrapper;
    beforeEach(() => {
     
        wrapper = shallow (
            <Breweries  name={mockData[0].name}
                        id={mockData[0].FIELD1}
                        dataset={mockData}
                        starredBreweries={mockData}
                        updateStarredList={updateStarredListMock}
                        key={0}
            />
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should have default state', () => {
        expect(wrapper.state()).toEqual({"beerList": [{"FIELD1": 43, "abv": 0.045, "brewery_id": 60, "ibu": 18, "id": 2544, "name": "Send Help", "ounces": 12, "style": "American Blonde Ale"}, {"FIELD1": 44, "abv": 0.055, "brewery_id": 60, "ibu": null, "id": 2324, "name": "Cast Iron Oatmeal Brown", "ounces": 12, "style": "American Brown Ale"}, {"FIELD1": 45, "abv": 0.06, "brewery_id": 60, "ibu": null, "id": 2288, "name": "Reprise Centennial Red", "ounces": 12, "style": "American Amber / Red Ale"}, {"FIELD1": 46, "abv": 0.055, "brewery_id": 60, "ibu": null, "id": 2287, "name": "Alter Ego", "ounces": 12, "style": "American Black Ale"}], "breweryName": "The Lion Brewery"})
    });

    it('Should toggle its favorites', () => {
      wrapper.instance().toggleFavorite();
      let add = 'add'
      expect(updateStarredListMock).toHaveBeenCalledWith(mockData[0].FIELD1, add)
    });

    it('Should set the state of beerList', () => {
      expect(wrapper.state('beerList')).toEqual(mockData[0].beers);
      wrapper.instance().getBeers();
      expect(wrapper.state('beerList')).toEqual(mockData[0].beers)
    });

    it('Should toggleBeerList when clicked', () => {
      const mock = jest.fn()
      const mockEvent = {
        target:{
          closest: ()=>({
            querySelector: () => ({
              classList: {toggle: mock}
            })
        })
      }
    }

      wrapper.find('.brewery-name').simulate('click', mockEvent);
      expect(mock).toHaveBeenCalled();
    });

    it('Should toggleFavorites when clicked', ()=> {
      
      wrapper.find('#star-filler').simulate('click');
      expect(updateStarredListMock).toHaveBeenCalled();
    })

});

