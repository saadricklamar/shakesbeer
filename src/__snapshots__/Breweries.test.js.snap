// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breweries Should match the snapshot 1`] = `
<div
  className="brewery-label"
>
  <div
    className="brewery-header"
  >
    <i
      className="far fa-star"
      id="star-filler"
      onClick={[Function]}
    />
    <h2
      className="brewery-name"
      onClick={[Function]}
    >
      The Lion Brewery
    </h2>
  </div>
  <div
    className="beer-dropdown hidden"
  >
    <Beer
      beerName="Send Help"
      beers={
        Array [
          Object {
            "FIELD1": 43,
            "abv": 0.045,
            "brewery_id": 60,
            "ibu": 18,
            "id": 2544,
            "name": "Send Help",
            "ounces": 12,
            "style": "American Blonde Ale",
          },
          Object {
            "FIELD1": 44,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2324,
            "name": "Cast Iron Oatmeal Brown",
            "ounces": 12,
            "style": "American Brown Ale",
          },
          Object {
            "FIELD1": 45,
            "abv": 0.06,
            "brewery_id": 60,
            "ibu": null,
            "id": 2288,
            "name": "Reprise Centennial Red",
            "ounces": 12,
            "style": "American Amber / Red Ale",
          },
          Object {
            "FIELD1": 46,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2287,
            "name": "Alter Ego",
            "ounces": 12,
            "style": "American Black Ale",
          },
        ]
      }
      key="43"
    />
    <Beer
      beerName="Cast Iron Oatmeal Brown"
      beers={
        Array [
          Object {
            "FIELD1": 43,
            "abv": 0.045,
            "brewery_id": 60,
            "ibu": 18,
            "id": 2544,
            "name": "Send Help",
            "ounces": 12,
            "style": "American Blonde Ale",
          },
          Object {
            "FIELD1": 44,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2324,
            "name": "Cast Iron Oatmeal Brown",
            "ounces": 12,
            "style": "American Brown Ale",
          },
          Object {
            "FIELD1": 45,
            "abv": 0.06,
            "brewery_id": 60,
            "ibu": null,
            "id": 2288,
            "name": "Reprise Centennial Red",
            "ounces": 12,
            "style": "American Amber / Red Ale",
          },
          Object {
            "FIELD1": 46,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2287,
            "name": "Alter Ego",
            "ounces": 12,
            "style": "American Black Ale",
          },
        ]
      }
      key="44"
    />
    <Beer
      beerName="Reprise Centennial Red"
      beers={
        Array [
          Object {
            "FIELD1": 43,
            "abv": 0.045,
            "brewery_id": 60,
            "ibu": 18,
            "id": 2544,
            "name": "Send Help",
            "ounces": 12,
            "style": "American Blonde Ale",
          },
          Object {
            "FIELD1": 44,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2324,
            "name": "Cast Iron Oatmeal Brown",
            "ounces": 12,
            "style": "American Brown Ale",
          },
          Object {
            "FIELD1": 45,
            "abv": 0.06,
            "brewery_id": 60,
            "ibu": null,
            "id": 2288,
            "name": "Reprise Centennial Red",
            "ounces": 12,
            "style": "American Amber / Red Ale",
          },
          Object {
            "FIELD1": 46,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2287,
            "name": "Alter Ego",
            "ounces": 12,
            "style": "American Black Ale",
          },
        ]
      }
      key="45"
    />
    <Beer
      beerName="Alter Ego"
      beers={
        Array [
          Object {
            "FIELD1": 43,
            "abv": 0.045,
            "brewery_id": 60,
            "ibu": 18,
            "id": 2544,
            "name": "Send Help",
            "ounces": 12,
            "style": "American Blonde Ale",
          },
          Object {
            "FIELD1": 44,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2324,
            "name": "Cast Iron Oatmeal Brown",
            "ounces": 12,
            "style": "American Brown Ale",
          },
          Object {
            "FIELD1": 45,
            "abv": 0.06,
            "brewery_id": 60,
            "ibu": null,
            "id": 2288,
            "name": "Reprise Centennial Red",
            "ounces": 12,
            "style": "American Amber / Red Ale",
          },
          Object {
            "FIELD1": 46,
            "abv": 0.055,
            "brewery_id": 60,
            "ibu": null,
            "id": 2287,
            "name": "Alter Ego",
            "ounces": 12,
            "style": "American Black Ale",
          },
        ]
      }
      key="46"
    />
  </div>
</div>
`;
